jobs/haproxy/spec

```diff
@@ -27,6 +27,9 @@ properties:
   ha_proxy.internal_only_domains:
     description: "Array of domains for internal-only apps/services (not hostnames for the apps/services)"
     default: []
+  ha_proxy.trusted_domain_cidrs: 
+    description: "Space separated trusted cidr blocks for internal_only_domains"
+    default: 0.0.0.0/0
```

Comment:
0.0.0.0/0 is allowing all source ip's by default.
http-request deny if internal !private

deny if internal domain and if source is not trusted_domain_cidrs.